src/main/scala/org/broadinstitute/dsde/firecloud/EntityService.scala

```diff
@@ -220,36 +220,47 @@ class EntityService(rawlsDAO: RawlsDAO, importServiceDAO: ImportServiceDAO, cwds
 
-    // generate unique name for the file-to-upload
-    val fileToWrite = GcsObjectName(s"incoming/${java.util.UUID.randomUUID()}.json")
-    val bucketToWrite = if (useCWDS) GcsBucketName(FireCloudConfig.Cwds.bucket) else GcsBucketName(FireCloudConfig.ImportService.bucket)
-
-    // write rawlsCalls to import service's bucket
     val dataBytes = rawlsCalls.toJson.prettyPrint.getBytes(StandardCharsets.UTF_8)
-    val insertedObject = googleServicesDAO.writeObjectAsRawlsSA(bucketToWrite, fileToWrite, dataBytes)
-    val gcsPath = s"gs://${insertedObject.bucketName.value}/${insertedObject.objectName.value}"
-
-    val importRequest = AsyncImportRequest(gcsPath, FILETYPE_RAWLS, Some(ImportOptions(None, Some(isUpsert))))
 
     if (useCWDS) {
-      importToCWDS(workspaceNamespace, workspaceName, userInfo, importRequest)
+      getWorkspaceId(workspaceNamespace, workspaceName, userInfo) map { workspaceId =>
+        val bucketToWrite = GcsBucketName(FireCloudConfig.Cwds.bucket)
+        val fileToWrite = GcsObjectName(s"to-cwds/${workspaceId}/${java.util.UUID.randomUUID()}.json")
+        val gcsPath = writeDataToGcs(bucketToWrite, fileToWrite, dataBytes)
+        val importRequest = getImportRequest(gcsPath, isUpsert)
+        importToCWDS(workspaceNamespace, workspaceName, workspaceId, userInfo, importRequest)
+      }
     } else {
+      val bucketToWrite = GcsBucketName(FireCloudConfig.ImportService.bucket)
+      val fileToWrite = GcsObjectName(s"incoming/${java.util.UUID.randomUUID()}.json")
+      val gcsPath = writeDataToGcs(bucketToWrite, fileToWrite, dataBytes)
+      val importRequest = getImportRequest(gcsPath, isUpsert)
       importServiceDAO.importJob(workspaceNamespace, workspaceName, importRequest, isUpsert)(userInfo)
     }
   }
 
-  private def importToCWDS(workspaceNamespace: String, workspaceName: String, userInfo: UserInfo, importRequest: AsyncImportRequest
-                          ): Future[PerRequestMessage] = {
-      // translate the workspace namespace/name into an id
-      rawlsDAO.getWorkspace(workspaceNamespace, workspaceName)(userInfo) map { workspace =>
-      // create the job in cWDS
-      val cwdsJob = cwdsDAO.importV1(workspace.workspace.workspaceId, importRequest)(userInfo)
-      // massage the cWDS job into the response format Orch requires
-      val asyncImportResponse = AsyncImportResponse(url = importRequest.url,
-        jobId = cwdsJob.getJobId.toString,
-        workspace = WorkspaceName(workspaceNamespace, workspaceName))
-      RequestComplete(Accepted, asyncImportResponse)
-    }
+  private def writeDataToGcs(bucket: GcsBucketName, objectName: GcsObjectName, data: Array[Byte]): String = {
+    val insertedObject = googleServicesDAO.writeObjectAsRawlsSA(bucket, objectName, data)
+    s"gs://${insertedObject.bucketName.value}/${insertedObject.objectName.value}"
+  }
+
+  private def getImportRequest(gcsPath: String, isUpsert: Boolean): AsyncImportRequest = {
```

Comment:
Since this class also deals with other import types, maybe this method could have a more specific name?

```suggestion
  private def getRawlsJsonImportRequest(gcsPath: String, isUpsert: Boolean): AsyncImportRequest = {
```

Reply:
Good call.  Pushed a commit to fix this.